### PR TITLE
Data sets arg format and testing for DataAccessFactory local

### PR DESCRIPTION
**Pull request by daw3rd**
## Why are these changes needed?
1. To unify the way data_data_sets and data_files_to_use are specified (as python/json lists of strings)
2. Add testing of DataAccessFactory and its creation of DataAccessLocal

## Related issue number (if any).

None

### CODE REVIEW

File: data-processing-lib/src/data_processing/data_access/data_access_local.py [hunk 1]
```diff
@@ -70,7 +70,7 @@ def get_output_folder(self) -> str:
         return self.output_folder
 
     @staticmethod
-    def _get_all_files_ext(path: str, extensions: list[str]) -> list[str]:
+    def _get_all_files_ext(path: str, extensions: list[str], absolute_paths: bool = True) -> list[str]:
         """
```
Comment (blublinsky) [inline data-processing-lib/src/data_processing/data_access/data_access_local.py]: This seems wrong. We always need absolute paths

File: data-processing-lib/src/data_processing/data_access/data_access_local.py [hunk 2]
```diff
@@ -80,11 +80,14 @@ def _get_all_files_ext(path: str, extensions: list[str]) -> list[str]:
...
             s_path = str(c_path.absolute())
-            if extensions is None:
-                if c_path.is_dir():
-                    continue
-            else:
+            if not absolute_paths:
+                s_path = s_path.replace(path, "")
+                if s_path.startswith(os.sep):
+                    s_path = s_path[1:]
+            if extensions is not None:
```
Comment (blublinsky) [inline data-processing-lib/src/data_processing/data_access/data_access_local.py]: Once again, we always need absolute paths

File: data-processing-lib/src/data_processing/data_access/data_access_local.py [hunk 3]
```diff
@@ -145,8 +148,10 @@ def _get_input_files(
                 path=input_path, cm_files=cm_files, min_file_size=min_file_size, max_file_size=max_file_size
             )
 
-        input_files = set(self._get_all_files_ext(path=input_path, extensions=self.files_to_use))
-        output_files = set(self._get_all_files_ext(path=output_path, extensions=self.files_to_use))
+        input_files = set(self._get_all_files_ext(path=input_path, extensions=self.files_to_use, absolute_paths=False))
+        output_files = set(
+            self._get_all_files_ext(path=output_path, extensions=self.files_to_use, absolute_paths=False)
+        )
         input_only_files = input_files - output_files
```
Comment (blublinsky) [inline data-processing-lib/src/data_processing/data_access/data_access_local.py]: This is completely wrong. we should always use absolute paths, as to different directories can have a file with the same name

File: data-processing-lib/src/data_processing/data_access/data_access_local.py [hunk 4]
```diff
@@ -155,8 +160,9 @@ def _get_input_files(
         for filename in sorted(input_only_files):
             if i >= cm_files > 0:
                 break
+            filename = os.path.join(input_path, filename)
             output_files.append(filename)
```
Comment (blublinsky) [inline data-processing-lib/src/data_processing/data_access/data_access_local.py]: Once again its wrong

**Review by yuanchi2807 — APPROVED** [review]
LGTM

**Review by yuanchi2807 — APPROVED** [review]

**Comment by blublinsky** [comment]
Fix for checkpointing is here https://github.com/IBM/data-prep-lab/pull/83

**Review by blublinsky — APPROVED** [review]
LGTM

**Pull request merged**